warehouse/packaging/models.py

```diff
@@ -135,6 +136,10 @@ class Project(SitemapMixin, db.Model):
         passive_deletes=True,
     )
 
+    events = orm.relationship(
+        "ProjectEvent", backref="project", cascade="all, delete-orphan", lazy=False
```

Comment:
Similar concern as the User/UserEvent relationship. We likely want to maintain the log post project deletion for the retention period.